dashboard/app/models/teacher_feedback.rb

```diff
@@ -57,23 +60,45 @@ def get_script_level
     script_level
   end
 
-  def self.get_student_level_feedback(student_id, level_id, teacher_id, script_id)
+  def self.get_student_level_feedback_by_teacher(student_id, level_id, script_id, teacher_id)
     where(
       student_id: student_id,
       level_id: level_id,
-      teacher_id: teacher_id,
-      script_id: script_id
+      script_id: script_id,
+      teacher_id: teacher_id
     ).latest
   end
 
-  def self.get_all_feedback_for_section(student_ids, level_ids, teacher_id)
-    find(
-      where(
-        student_id: student_ids,
-        level_id: level_ids,
-        teacher_id: teacher_id
-      ).group([:student_id, :level_id]).pluck('MAX(teacher_feedbacks.id)')
-    )
+  def self.get_all_student_level_feedback(student_id, level_id, script_id)
+    where(
+      student_id: params.require(:student_id),
+      level_id: params.require(:level_id),
+      script_id: params.require(:script_id)
+    ).latest_per_teacher
+  end
+
+  def self.student_has_feedback(student_id)
+    where(student_id: student_id).count > 0
+  end
+
+  def self.get_student_unseen_feedback_count(student_id)
+    authorized_unseen_feedbacks = where(
+      student_id: current_user.id,
+      seen_on_feedback_page_at: nil,
+      student_first_visited_at: nil
+    ).select do |feedback|
+      feedback.teacher.authorized_teacher?
+    end
+
+    authorized_unseen_feedbacks.count
+  end
+
+  def self.get_student_feedbacks_for_script_by_teacher(script_id, student_ids, teacher_id = nil)
```

Comment:
How is this method and the one on 63 different?  When would they return different records?

Reply:
The one on 63 returns latest per teacher, it may return multiple feedbacks per level if they're from different teachers, whereas this method will just return the latest feedback per level

Reply:
When would we want to use one vs. the other?

Reply:
Based on how TopInstruction was written, I'm assuming that sometime in the past we showed more than one feedback to the student on a level if they had feedback from different teachers. We only show one feedback now, so I don't think it's necessary to query for the latest from all teachers. I've removed that query from the next refactor